src/probe/probe.c

```diff
@@ -1012,6 +1012,11 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 			}
 			/* ignore probe stream tag for extraction probes */
 			probe[i].stream_tag = _probe->ext_dma.stream_tag;
+		} else {
+			trace_probe_error("probe_point_add() error: invalid purpose %d",
+					  probe[i].purpose);
+
```

Comment:
although valid here, I'd think that this is the first sanity check that the function should be doing. we can simply check if probe[i].purpose > PROBE_PURPOSE_EXTRACTION and return invalid.

Also, the other concern I have is, Because there's the provision for adding multiple probe points with the same command, whats the logic here, do you add all "valid" probe points and ignore the "invalid" ones or you add the first "valid" probe points and return an error as soon as you encounter the first invalid input?

Reply:
Good point, moved to the beginning.

Currently the logic is to stop and return error if the input is invalid, there are several returns in this function in the case of error.
Of course this can be changed but it should be requested for another PR.

Reply:
Thanks @abonislawski. I will file an issue for it.